src/prisma/schema.prisma

```diff
@@ -0,0 +1,93 @@
+// This is your Prisma schema file,
+// learn more about it in the docs: https://pris.ly/d/prisma-schema
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+
+model User {
+  id                     String         @id @default(cuid())
+  email                  String?        @unique
+  address                String         @unique
+  listings               Listing[]
+  auctions               Auction[]
+  initiatedListingOffers ListingOffer[] @relation("initiatedListingOffers")
+  receievedListingOffers ListingOffer[] @relation("receivedListingOffers")
+  initiatedAuctionBids   AuctionBid[]   @relation("initiatedAuctionBids")
+  receivedAuctionBids    AuctionBid[]   @relation("receivedAuctionBids")
```

Comment:
```suggestion
  listings               Listing[]
  auctions               Auction[]
  initiatedListingOffers ListingOffer[] @relation("initiatedListingOffers")
  initiatedAuctionBids   AuctionBid[]   @relation("initiatedAuctionBids")
```
Since every listing created means you cannot initiate an offer, we can easily query for `receivedOffers` or `receivedBids`. If you think it won't be easy/convenient, we can just stick to what you currently have.